omegaconf/_utils.py

```diff
@@ -201,6 +201,37 @@ def _resolve_forward(type_: Type[Any], module: str) -> Type[Any]:
         return type_
 
 
+def extract_dict_subclass_data(obj: Any, parent: Any) -> Optional[Dict[str, Any]]:
+    """Check if obj is an instance of a subclass of Dict. If so, extract the Dict keys/values."""
```

Comment:
How does this interact with class fields?
```python
@dataclass
class DictSubclassWithField(Dict[str, Any]):
  age: int = 10

cfg = OmegaConf.create(DictSubclassWithField)
cfg.age = 11
cfg.name = "Commander Keen"

cfg2 = OmegaConf.create(cfg)
assert cfg2 == cfg
```

Reply:
> How does this interact with class fields?

I don't think there is a change in behavior for class fields.

The only difference should be that on this branch we have
```python
src = DictSubclassWithField(age=11)
src["name"] = "Commander Keen"
print(OmegaConf.create(src))  # prints {'age': 11, 'name': 'Commander Keen'}
```
and on the master branch we have
```python
src = DictSubclassWithField(age=11)
src["name"] = "Commander Keen"
print(OmegaConf.create(src))  # prints {'age': 11}
```